projects/bloomreach-experience-ng-sdk/src/lib/common-sdk/utils/page-model.ts

```diff
@@ -34,6 +35,12 @@ export function _buildApiUrl(apiUrls: ApiUrls, preview: boolean, urlPath: string
   if (componentId) {
     url = addComponentRenderingURL(url, componentId, apiUrls);
   }
+
+  const [, ssoHandshake = ''] = (query && query.match(SSO_HANDSHAKE)) || [];
+  if (ssoHandshake) {
+    url += (url.indexOf('?') === -1 ? '?' : '&') + ssoHandshake;
```

Comment:
Why not introduce a function like `addQueryParameter(url, parameter)` that can be reused here and in `addComponentRenderingUrl`? And too bad we don't have unit tests, it would be great to cover all this.

Reply:
Can't we use the `URLSearchParams` object for all the query string parsing and interpretation? That would make the code a lot more readable and remove the need for the SSO_HANDSHAKE regular expression.

Reply:
It will require extra polyfill to keep browser compatibility. Let's keep it until the next major release.